extras/ai-battle/main.cpp

```diff
@@ -0,0 +1,127 @@
+        ("replay", po::value(&replay_path),"Filename to write replay to (optional)")
+        ("save", po::value(&savegame_path),"Filename to write savegame to (optional)")
+        ("random_init", po::value<unsigned>()->default_value(std::chrono::high_resolution_clock::now().time_since_epoch().count()),"Seed value for the random number generator (optional)")
+        ("maxGF", po::value<unsigned>()->default_value(std::numeric_limits<unsigned>::max()),"Maximum number of game frames to run (optional)")
+        ("version", "Show version information and exit")
+        ;
+    // clang-format on
+
+    if(argc == 1)
+    {
+        bnw::cerr << desc << std::endl;
+        return 1;
+    }
+
+    po::variables_map options;
+    try
+    {
+        po::store(po::command_line_parser(argc, argv).options(desc).run(), options);
+
+        if(options.count("help"))
+        {
+            bnw::cout << desc << std::endl;
+            return 0;
+        }
+        if(options.count("version"))
+        {
+            bnw::cout << rttr::version::GetTitle() << " v" << rttr::version::GetVersion() << "-"
+                      << rttr::version::GetRevision() << std::endl
+                      << "Compiled with " << System::getCompilerName() << " for " << System::getOSName() << std::endl;
+            return 0;
+        }
+
+        po::notify(options);
+    } catch(const std::exception& e)
+    {
+        bnw::cerr << "Error: " << e.what() << std::endl;
+        bnw::cerr << desc << std::endl;
+        return 1;
+    }
+
+    try
+    {
+        auto random_init = options["random_init"].as<unsigned>();
+
+        // We print arguments and seed in order to be able to reproduce crashes.
+        for(int i = 0; i < argc; ++i)
+            bnw::cout << argv[i] << " ";
+        bnw::cout << std::endl;
+        bnw::cout << "random_init: " << random_init << std::endl;
+        bnw::cout << std::endl;
+
+        RTTRCONFIG.Init();
+        RANDOM.Init(random_init);
+
+        const bfs::path mapPath = RTTRCONFIG.ExpandPath(options["map"].as<std::string>());
+        const std::vector<AI::Info> ais = ParseAIOptions(options["ai"].as<std::vector<std::string>>());
+
+        GlobalGameSettings ggs;
+        const auto objective = options["objective"].as<std::string>();
+        if(objective == "domination")
+            ggs.objective = GameObjective::TotalDomination;
+        else if(objective == "conquer")
+            ggs.objective = GameObjective::Conquer3_4;
+        else
+        {
+            bnw::cerr << "unknown objective: " << objective << std::endl;
+            return 1;
+        }
+
+        ggs.objective = GameObjective::TotalDomination;
+        HeadlessGame game(ggs, mapPath, ais);
+        if(replay_path)
+            game.RecordReplay(*replay_path, random_init);
+
+        unsigned maxGF = options["maxGF"].as<unsigned>();
+
+        game.Run(maxGF);
```

Comment:
Either make it `const` or use it directly:
```suggestion
        game.Run(options["maxGF"].as<unsigned>());
```

Reply:
done